builder/remote_test.go

```diff
@@ -144,3 +153,47 @@ func TestInspectResponseEmptyContentType(t *testing.T) {
 		t.Fatalf("Corrupted response body %s", body)
 	}
 }
+
+func TestMakeRemoteContext(t *testing.T) {
+	contextDir, err := ioutil.TempDir("", "builder-remote-test")
+
+	if err != nil {
+		t.Fatalf("Error with creating temporary directory: %s", err)
+	}
+
+	defer os.RemoveAll(contextDir)
+
+	testFilename := filepath.Join(contextDir, DefaultDockerfileName)
+	err = ioutil.WriteFile(testFilename, []byte(textPlainDockerfile), 0777)
+
+	if err != nil {
+		t.Fatalf("Error when writing file (%s) contents: %s", testFilename, err)
+	}
+
+	mux := http.NewServeMux()
+	server := httptest.NewServer(mux)
+	serverURL, _ := url.Parse(server.URL)
+
+	serverURL.Path = "/" + DefaultDockerfileName
+	remoteURL := serverURL.String()
+
+	mux.Handle("/", http.FileServer(http.Dir(contextDir)))
+
+	remoteContext, err := MakeRemoteContext(remoteURL, map[string]func(io.ReadCloser) (io.ReadCloser, error){
+		httputils.MimeTypes.TextPlain: func(rc io.ReadCloser) (io.ReadCloser, error) {
+			dockerfile, err := ioutil.ReadAll(rc)
+			if err != nil {
+				return nil, err
+			}
+			return archive.Generate(DefaultDockerfileName, string(dockerfile))
+		},
+	})
+
+	if err != nil {
+		t.Fatalf("Error when executing DetectContextFromRemoteURL: %s", err)
+	}
+
+	if remoteContext == nil {
```

Comment:
Don't you want to test more on context? Like, try to cast it to `*tarSumContext` and inspect fields.